### DISCUSSION TITLE
How to turn of the light casting shadow

### DISCUSSION THREAD

**Discussion by HSDSZ** [bnpr/Malt#466] [Q&A]
Like the function in eevee, If i turn the swtich off, the shadow disappear. Any way to achieve this function in malt light settingn?
![image](https://user-images.githubusercontent.com/36077492/210844143-66e49603-9a6d-476f-9592-fcff98618054.png)

![image](https://user-images.githubusercontent.com/36077492/210844438-04c425a6-e9ed-445c-b78f-fc14d5a050f7.png)

**Comment by pragma37** [comment]
While I think we should add this as a built-in feature, you can get the same result using Light Groups.
If you set the Sun Light Group to 2, for example, you can set your Materials Light Groups like this:
![imagen](https://user-images.githubusercontent.com/36610999/210845275-53d8b10a-03b5-4b2d-b38b-bba8ad2b7f81.png)
So this Material receives light from Light Group 2 but doesn't cast shadows in it.
Lights in Light Group 1 will still work as normal.

**Reply by HSDSZ** [reply]
I am also doing this as a work around. But I have a lot of assets, it is hard to go back and give every material an extra light number.

I am wondering is it possible to let all material always use an internal light group and shadow group number, such as -999.

If i want a light to turn off casting shadow, I just add a new light, set the light group to -999, shadow group to other, then it will work.

**Reply by pragma37** [reply]
You can run a script that sets the Light Group for you.
So, if your "no shadows" group is `-1` you can run this script from the Blender Text Editor:
``` python
import bpy
for material in bpy.data.materials:
    if "Light Groups.Light" in material.malt_parameters.keys():
        material.malt_parameters["Light Groups.Light"][3] = -1
````
If you check the `Register` option it will run automatically when you load the file.

**Reply by HSDSZ** [reply]
This script does not work for linked material. Although it changes the linked light group socket.
![image](https://user-images.githubusercontent.com/36077492/211052623-57dc019a-723c-47a7-a919-7b96fd89c009.png)

**Reply by HSDSZ** [reply]
Just got a way.

I changed the pipeline.py.  ^_^
set the default light group to -99 and shadow group to -98. Then I could have light that only cast shadow or only light up
![image](https://user-images.githubusercontent.com/36077492/211057969-cae488b0-ead6-494b-ab1f-3ea9036550f6.png)
